src/shell/js/composeui-fdc3/src/ComposeUITopic.ts

```diff
@@ -0,0 +1,21 @@
+/* 
+ *  Morgan Stanley makes this available to you under the Apache License,
+ *  Version 2.0 (the "License"). You may obtain a copy of the License at
+ *       http://www.apache.org/licenses/LICENSE-2.0.
+ *  See the NOTICE file distributed with this work for additional information
+ *  regarding copyright ownership. Unless required by applicable law or agreed
+ *  to in writing, software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ *  or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ *  
+ */
+
+export class ComposeUITopic {
+    private static readonly topicRoot = "composeui/fdc3/v2.0/";
+    private static readonly userChannelTopicRoot: string = this.topicRoot + "userchannels/";
```

Comment:
The concatenation of strings is correct, but there's a way that makes them more readable in the long run, since a topic is is basically substring1/substring2/...

I'd remove the closing / from the end of topicRoot and construct the formatted strings the following way with Template literal strings:

`` private static readonly userChannelTopicRoot: string = `${topicRoot}/userchannels/` ``

same for line 19.

Optionally you can also create const variables for the "userchannels" and "broadcast" strings, the readability could improve even more, and all the constants would be in one place, but only if we'll reuse those:

`${topicRoot}/${USER_CHANNELS}/`

I'd also put "v2.0" in a variable too for future improvements.

Reply:
Sorry, I have to nitpick, somehow the current solution looks like "concatenation with extra steps" instead of leveraging what template strings are for.